vnext/Desktop/WinHTTPWebSocketResource.cpp

```diff
@@ -0,0 +1,447 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#include "pch.h"
+
+#include <Unicode.h>
+#include <Utils.h>
+#include "WinHTTPWebSocketResource.h"
+
+// Windows API
+#include <comutil.h>
+
+using Microsoft::Common::Unicode::Utf8ToUtf16;
+
+using std::function;
+using std::size_t;
+using std::string;
+using std::wstring;
+
+namespace Microsoft::React
+{
+
+WinHTTPWebSocketResource::WinHTTPWebSocketResource(const string& urlString)
+  : m_openSessionFlags{ WINHTTP_FLAG_ASYNC }
+  , m_sessionHandle{ NULL }
+  , m_connectionHandle{ NULL }
+  , m_requestHandle{ NULL }
+  , m_errorStatus{ ERROR_SUCCESS }
```

Comment:
consider moving these trivial ones to the header
please use nullptr instead of NULL

Reply:
Certainly.

Please note, this class will remain for experimental purposes, but will not be exported to the DLL for now.

The main class to review is `WinRTWebSocketResource`.